src/main/java/pl/psnc/dei/iiif/Converter.java

```diff
@@ -201,8 +211,13 @@ private void replaceUrl(ConversionDataHolder.ConversionData data) throws Malform
 	 * @param file file to which save contents
 	 * @throws IOException
 	 */
-	private void copyURLToFile(URL url, File file) throws IOException {
-		HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+	private void copyURLToFile(URL url, File file) throws IOException, DownloadFileException {
+		HttpURLConnection connection;
+		try {
+			connection = (HttpURLConnection) url.openConnection();
+		} catch (ClassCastException e) {
```

Comment:
Why do you catch here CastClassException? It's a bit strange for me especially that it actually informs that casting to HttpURLConnection failed, not that the downloaded file does not exist...

Reply:
Refactored this to be more readable